src/test/java/commands/AddSalesCommandTest.java

```diff
@@ -0,0 +1,85 @@
+package commands;
+
+import author.Author;
+import author.AuthorList;
+import exceptions.TantouException;
+import manga.Manga;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import sales.Sale;
+import ui.Ui;
+
+import java.io.PrintStream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.fail;
+
+//@@author sarahchow03
+class AddSalesCommandTest {
+    private final PrintStream standardOut = System.out;
+    private AuthorList authorList;
+    private Ui ui;
+    private Author existingAuthor;
+    private Manga existingManga;
+    private AddSalesCommand commandUnderTest;
+
+    @BeforeEach
+    public void setUp() {
+        authorList = new AuthorList();
+        ui = new Ui();
+        existingAuthor = new Author("test author");
+        existingManga = new Manga("test manga", existingAuthor);
+        existingAuthor.addManga(existingManga);
+        authorList.add(existingAuthor);
+    }
+
+    @Test
+    public void execute_validInput_addSalesDataSuccessfully() {
+        try {
+            commandUnderTest = new AddSalesCommand("sales -a \"test author\" -m \"test manga\" -p \"10.90\" -q \"20\"");
+            commandUnderTest.execute(ui, authorList);
+            Sale sale = existingAuthor.getManga("test manga").getSalesData();
+            assertEquals(20, sale.getQuantitySold());
+            assertEquals(10.90, sale.getUnitPrice());
+        } catch (TantouException e) {
+            fail();
+        } finally {
+            System.setOut(standardOut);
```

Comment:
Is this finally block needed? It seems like `setOut()` is not changed anywhere else in the class.